src/components/ClusterManager.js

```diff
@@ -194,17 +196,19 @@ export default class ClusterManager extends PureComponent {
     return currentCluster(clusters)
   }
 
-  async executeAndRefresh(promise) {
-    try {
-      const { refreshClusters } = this.props
-      this.setState({ busy: true })
-      await promise
-      await refreshClusters()
-    } catch (error) {
-      reportError('Notebook Runtime Error', error)
-    } finally {
-      this.setState({ busy: false })
-    }
+  async executeAndRefresh(promise, waitBeforeRefreshMillis = 0) {
+    await setTimeout(async () => {
```

Comment:
did I do this timeout + await with nested async await logic correctly? Everything checked out in my testing.